app/parsers/bulkrax/csv_parser.rb

```diff
@@ -52,14 +54,32 @@ def create_works
         break if !limit.nil? && index >= limit
 
         seen[record[:source_identifier]] = true
-        new_entry = entry_class.where(importer: importer, identifier: record[:source_identifier], raw_metadata: record.to_h.compact).first_or_create!
-        ImportWorkJob.perform_later(new_entry.id, importer.current_importer_run.id)
+        new_entry = find_or_create_entry(entry_class, record[:source_identifier], 'Bulkrax::Importer', record.to_h.compact)
+        ImportWorkJob.perform_later(new_entry.id, current_importer_run.id)
         increment_counters(index)
       end
     rescue StandardError => e
       errors.add(:base, e.class.to_s.to_sym, message: e.message)
     end
 
+    def create_from_importer
+      importer = Bulkrax::Importer.find(importerexporter.export_source)
```

Comment:
I think that this and its sister method below are what should be done by the ExportJob?